storage/store/chunk_data_packs.go

```diff
@@ -13,118 +15,251 @@ import (
+//     storage layer, we _could_ in phase 1 store all known chunk data packs. However, an Execution Node may only commit to a single
+//     chunk data pack (or it will get slashed). This mapping from chunk ID to the ID of the chunk data pack that the Execution Node
+//     actually committed to is stored in the protocol database, in the following phase 2.
+//   - In the second phase, we populate the index mappings from ChunkID to one "distinguished" chunk data pack ID. This mapping
+//     is stored in the protocol database. Typically, en Execution Node uses this for indexing its own chunk data packs which it
+//     publicly committed to.
+//   - This function can approximately be described as an atomic operation. When it completes successfully, either both databases
+//     have been updated, or neither. However, this is an approximation only, because interim states exist, where the chunk data
+//     packs already have been stored in the chunk data pack database, but the index mappings do not yet exist.
+//
+// The Store method returns:
+//   - func(lctx lockctx.Proof, rw storage.ReaderBatchWriter) error: Function for populating the index mapping from chunkID
+//     to chunk data pack ID in the protocol database. This mapping persists that the Execution Node committed to the result
+//     represented by this chunk data pack. This function returns [storage.ErrDataMismatch] when a _different_ chunk data pack
+//     ID for the same chunk ID has already been stored (changing which result an execution Node committed to would be a
+//     slashable protocol violation). The caller must acquire [storage.LockInsertChunkDataPack] and hold it until the database
+//     write has been committed.
+//   - error: No error should be returned during normal operation. Any error indicates a failure in the first phase.
+func (ch *ChunkDataPacks) Store(cs []*flow.ChunkDataPack) (
+	func(lctx lockctx.Proof, protocolDBBatch storage.ReaderBatchWriter) error,
+	error,
+) {
+
+	// Phase 1: Store chunk data packs in dedicated (separate) database. This converts the
+	// ChunkDataPacks to the reduced StoredChunkDataPacks representation and stores them.
+	storedChunkDataPacks := storage.ToStoredChunkDataPacks(cs)
+
+	// Store the chunk data packs and get back their unique IDs
+	chunkDataPackIDs, err := ch.stored.StoreChunkDataPacks(storedChunkDataPacks)
+	if err != nil {
+		return nil, fmt.Errorf("cannot store chunk data packs: %w", err)
+	}
+
+	// Sanity check: validate that we received the expected number of IDs
+	if len(cs) != len(chunkDataPackIDs) {
+		return nil, fmt.Errorf("stored chunk data pack IDs count mismatch: expected: %d, got: %d",
+			len(cs), len(chunkDataPackIDs))
+	}
+
+	// Phase 2: Create the function that will index chunkID -> chunkDataPackID mappings
+	storeChunkDataPacksFunc := func(lctx lockctx.Proof, protocolDBBatch storage.ReaderBatchWriter) error {
+		protocolDBBatch.AddCallback(func(err error) {
```

Comment:
[follow-up from slack conversation](https://flow-foundation.slack.com/archives/C074X15RQH1/p1760487997060739?thread_ts=1760487990.013869&cid=C074X15RQH1): I would suggest to remove this. 

```suggestion
```